Added long press menu in tweet list

siteshwar opened the pull request:
Fix for #57

@veskuh My changes are working fine except sometimes context menu appears over tweet instead of below it. Please suggest a fix for this.

siteshwar committed:
Added long press menu in tweet list

veskuh commented on qml/tweetian-harmattan/Delegate/TweetDelegate.qml:
```diff
@@ -21,6 +21,9 @@ import Sailfish.Silica 1.0
 
 AbstractDelegate {
     id: root
+    height: menuOpen ? root.implicitHeight + tweetView.contextMenu.height : root.implicitHeight
```
This may be reason for the random issues. One thing to experiment with would be to change AbstractDelegate from BackgroundItem to ListItem and instead of defining implicitHeight in it, we would define contentHeight. That's closer to how context menu should be defined.

siteshwar replied:
Changing BackgroundItem to ListItem and using contentHeight instead of implictHeight does not fix this problem (instead it screws up height of all the items and they start appearing over each other). It has something to do with the height of tweetView itself as if I scroll down, height of tweetView increases and context menu never appears over list items.

BTW Examples in https://sailfishos.org/sailfish-silica/qml-sailfishsilica-contextmenu.html use BackgroundItem instead of ListItem too.

veskuh commented on qml/tweetian-harmattan/Delegate/TweetDelegate.qml:
```diff
@@ -117,5 +120,11 @@ AbstractDelegate {
+
+    onPressAndHold: {
+        if (!tweetView.contextMenu)
+            tweetView.contextMenu = contextMenuComponent.createObject(tweetView)
```
All the items in list will share single instance of context menu. I'm not sure if it works correctly in all cases. Typically we have delegate itself owning the context menu.

siteshwar replied:
Example at https://sailfishos.org/sailfish-silica/qml-sailfishsilica-contextmenu.html uses shared context menu.

siteshwar replied:
Fixed through 666b00a

veskuh commented on qml/tweetian-harmattan/MainPageCom/TweetListView.qml:
```diff
@@ -171,6 +173,34 @@ Item {
         }
+
+        Component {
+            id: contextMenuComponent
```
There is already Dialog/LongPressMenu it should be fixed and used instead of defining these here (and then we get the existing translations).

siteshwar replied:
Fixed through 666b00a

siteshwar committed:
Use LongPressMenu component to show context menu

Moved context menu object inside delegate, every delegate has it's own context
menu now

veskuh commented:
I still see context menu sometimes over tweets. I think I have a fix. I'll merge this and push fixes on top.
